src/main/java/Parser.java

```diff
@@ -0,0 +1,169 @@
+public class Parser {
+    public void parse(String prompt, TaskList tasks, Ui ui, Storage storage) throws InvalidArgumentException, InvalidTaskException, InvalidSerialException {
+        if (prompt.equals("list")) {
+            ui.showLine();
+            System.out.println("Here are the tasks in your list:\n"
+                    + tasks.getTasks());
+            ui.showLine();
+        } else if (prompt.contains("mark") || prompt.contains("unmark") ||
+                prompt.contains("delete")) {
+            int index = Integer.parseInt(prompt.charAt(prompt.length() - 1) + "");
+            if(index > tasks.size()) {
+                throw new InvalidSerialException("☹ OOPS!!! I think you have added" +
+                        "an incorrect serial number greater than " + (tasks.size() - 1));
+            }
```

Comment:
```suggestion
            if(index > tasks.size()) {
                throw new InvalidSerialException("☹ OOPS!!! I think you have added"
                        + "an incorrect serial number greater than " + (tasks.size() - 1));
            }
```
Perhaps consider moving the operator to the next line to improve readability? I noticed the same issue in several other places too.

More details here: https://se-education.org/guides/conventions/java/intermediate.html#layout